test/distributed/org/apache/cassandra/fuzz/topology/TopologyMixupTestBase.java

```diff
@@ -464,19 +488,46 @@ public String ipAddress(int nodeNum)
                 result.asserts().success();
                 logger.info("CMS reconfigure: {}", result.getStdout());
             }
-            preActions.add(new Runnable()
-            {
+            commandsTransformers.add(new BiFunction<State<S>, Gen<Command<State<S>, Void, ?>>, Gen<Command<State<S>, Void, ?>>>() {
                 // in order to remove this action, an anonymous class is needed so "this" works, lambda "this" is the parent class
                 @Override
-                public void run()
-                {
-                    if (topologyHistory.up().length == TARGET_RF)
-                    {
+                public Gen<Command<State<S>, Void, ?>> apply(State<S> state, Gen<Command<State<S>, Void, ?>> commandGen) {
+                    if (topologyHistory.up().length < TARGET_RF)
+                        return commandGen;
+                    SimpleCommand<State<S>> reconfig = new SimpleCommand<>("nodetool cms reconfigure " + TARGET_RF, ignore -> {
                         NodeToolResult result = cluster.get(1).nodetoolResult("cms", "reconfigure", Integer.toString(TARGET_RF));
                         result.asserts().success();
                         logger.info("CMS reconfigure: {}", result.getStdout());
-                        preActions.remove(this);
-                    }
+                    });
+                    SimpleCommand<State<S>> fixDistributedSchemas = new SimpleCommand<>("Set system distributed keyspaces to RF=" + TARGET_RF, ignore ->
+                            fixDistributedSchemas(cluster));
+                    SimpleCommand<State<S>> fixTestKeyspace = new SimpleCommand<>("Set " + KEYSPACE + " keyspace to RF=" + TARGET_RF, s -> {
+                        cluster.schemaChange("ALTER KEYSPACE " + KEYSPACE + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': " + TARGET_RF + "}");
+                        rf = new TokenPlacementModel.SimpleReplicationFactor(TARGET_RF);
+                    });
+                    var self = this;
+                    return rs -> {
+                        Command<State<S>, Void, ?> next = commandGen.next(rs);
+                        if (next.checkPreconditions(state) == Property.PreCheckResult.Ignore)
+                            return next;
+                        commandsTransformers.remove(self);
+                        int[] up = state.topologyHistory.up();
+                        List<Command<State<S>, Void, ?>> commands = new ArrayList<>();
+                        commands.add(fixDistributedSchemas);
+                        for (String ks : Arrays.asList("system_auth", "system_traces"))
+                        {
+                            int coordinator = rs.pickInt(up);
+                            commands.add(repairCommand(coordinator, ks));
+                        }
```

Comment:
if there is data you *need* to repair to make things safe... so added the repairs for that reason